### PR TITLE
Feature/message doi prototype

### PR DESCRIPTION
**Pull request by denis-yuen**
A prototype for https://github.com/ga4gh/dockstore/issues/442 
Leaving open an interface that can be implemented if we decide to go with Datacite or Crossref directly
EventConsumer might be used for other time intensive, scalable tasks too.

Needs more testing in a future PR

### CODE REVIEW

File: dockstore-common/src/main/java/io/dockstore/common/model/DOIMessage.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package io.dockstore.common.model;
+
+public class DOIMessage extends BasicMessage {
+    private String targetEntry;
```
Comment (agduncan94) [inline dockstore-common/src/main/java/io/dockstore/common/model/DOIMessage.java]: What is the targetEntry?

Comment (denis-yuen) [reply dockstore-common/src/main/java/io/dockstore/common/model/DOIMessage.java]: The entry (tool or workflow) that should have a DOI requested for it. 
I'll add some Javadoc

File: dockstore-event-consumer/src/main/java/io/dockstore/consumer/EventConsumer.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+     *
+     * @param args
+     * @return
+     */
+    public static void main(String[] args) {
+
+        EventConsumer consumer = new EventConsumer();
+        String sqsURL = consumer.getConsumerConfiguration().getString("sqsURL");
+        String dockstoreToken = consumer.getConsumerConfiguration().getString("dockstoreToken");
+        String dockstoreURl = consumer.getConsumerConfiguration().getString("dockstoreURL");
```
Comment (garyluu) [inline dockstore-event-consumer/src/main/java/io/dockstore/consumer/EventConsumer.java]: Name vs string doesn't match

File: dockstore-event-consumer/src/main/java/io/dockstore/consumer/EventConsumer.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+import org.slf4j.LoggerFactory;
+
+public final class EventConsumer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EventConsumer.class);
+
+    /**
+     * the maximum wait time allowed
+     */
+    private static final int MAX_WAITTIME_SECONDS = 20;
```
Comment (garyluu) [inline dockstore-event-consumer/src/main/java/io/dockstore/consumer/EventConsumer.java]: Wait time possibly separated by underscore unless there's a limit to # of underscores

File: dockstore-event-consumer/src/main/java/io/dockstore/consumer/EventConsumer.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                        LOG.info("Deleting the message");
+                        sqs.deleteMessage(new DeleteMessageRequest(sqsURL, messageReceiptHandle));
+                    } else {
+                        // requeue the message
+                        sqs.changeMessageVisibility(sqsURL, messageReceiptHandle, 0);
+                    }
+                }
+
+            }
+        } while (true);
```
Comment (garyluu) [inline dockstore-event-consumer/src/main/java/io/dockstore/consumer/EventConsumer.java]: How often does this loop execute?

File: dockstore-webservice/src/main/resources/swagger.yaml [hunk 5]
```diff
@@ -5086,6 +5159,16 @@ definitions:
       verifiedSource:
         type: "string"
         description: "Verified source for the version"
+      doiURL:
```
Comment (garyluu) [inline dockstore-webservice/src/main/resources/swagger.yaml]: working_directory is snake case, verifiedSource is camel-case, do we have a set standard for non-ga4gh properties?

File: dockstore-event-consumer/src/main/java/io/dockstore/consumer/EventConsumer.java (hunk 4 above)
Comment (denis-yuen) [reply dockstore-event-consumer/src/main/java/io/dockstore/consumer/EventConsumer.java]: It is intended to consume messages ad infinitum till shutdown

Comment (garyluu) [reply dockstore-event-consumer/src/main/java/io/dockstore/consumer/EventConsumer.java]: I was more wondering which situations should we use some sort of polling rate vs just using infinite loop.

Comment (denis-yuen) [reply dockstore-event-consumer/src/main/java/io/dockstore/consumer/EventConsumer.java]: Depends, this prototype uses the maximimum AWS setting of 20 seconds to timeout per request. 
We can tinker to see if a different rate is more appropriate

**Pull request merged**